File: platform/shared/rhoruby/impl/RhoRubySingletone.cpp

#include "../api/RhoRubySingletone.h"
#include "RhoRubyImpl.h"
#include <stdlib.h>

namespace rho {
namespace ruby {

IRhoRuby* RhoRubySingletone::mRhoRuby = NULL;

IRhoRuby* RhoRubySingletone::getRhoRuby(bool lateInit) {
    if (mRhoRuby == NULL) {
        mRhoRuby = new RhoRubyImpl(lateInit);
    }
    return mRhoRuby;
}


}
}

File: platform/iphone/rhodesruby/impl/RhoRubySingletone.m

#import "../api/RhoRubySingletone.h"
#import "RhoRubyFabrique.h"


static id<IRhoRuby> ourIRhoRuby = nil;

@implementation RhoRubySingletone

+(id<IRhoRuby>) getRhoRuby {
    if (ourIRhoRuby == nil) {
        ourIRhoRuby = [RhoRubyFabrique createRhoRuby];
    }
    return ourIRhoRuby;
}

@end


